packages/framework/src/Form/JsFormValidatorFactory.php

```diff
@@ -62,7 +62,7 @@ protected function normalizeViewTransformers(FormInterface $form, array $viewTra
                 }
             }
 
-            array_push($transformer['choiceList'], $optionsItemsThatAreNotInstanceOfParameterValue);
+            $transformer['choiceList'] = array_merge($transformer['choiceList'], $optionsItemsThatAreNotInstanceOfParameterValue);
```

Comment:
from the previous PR https://github.com/shopsys/shopsys/pull/89/files this looks like it was implemented intentionally.
is this OK ? should not make more sense to remove the whole functionality?
example:
```
$transformer['choiceList'] == [1=>ParameterValue, 3=>ParameterValue] 
$optionsItemsThatAreNotInstanceOfParameterValue == [0=>Brand,1=>Flag,2=>Brand,3=>Flag]
array_push == [1=>ParameterValue, 3=>ParameterValue] , 4 => [0=>Brand,1=>Flag,2=>Brand,3=>Flag]]
array_merge == [0=>ParameterValue, 1=>ParameterValue] , 2=>Brand, 3=>Flag, 4=>Brand, 5=>Flag]
```
is also this change connected with the ProductFilterFormType? Should not be that solved in another commit?

Reply:
This change is related to the problem so i think it should be in this commit